src/components/NavBar/ChainSwitcher.tsx

```diff
@@ -107,7 +105,7 @@ export const ChainSwitcher = ({ leftAlign }: ChainSwitcherProps) => {
         background={isOpen ? 'accentActiveSoft' : 'none'}
         onClick={toggleOpen}
       >
-        {!isSupported ? (
+        {!isSupported || !info ? (
```

Comment:
there shouldn't be a case where you need to check !info

Reply:
and inside this you can always render Unsupported vs checking for info.label

Reply:
that was the point of this bug, Palm didn't have info so nothing was rendering. This also works as a null check for the info needed on the supported case

Reply:
if palm didn't have info wouldn't that be an unsupported chain?

Reply:
if we need to check `!info` for typescript reasons, could skip doing the `isSupported` check instead? they're functionally the same.

Reply:
```ts
  const isSupported = !!info

  const dropdown = (
    <NavDropdown top={54} leftAligned={leftAlign} paddingBottom={8} paddingTop={8}>
      <Column marginX="8">
        {NETWORK_SELECTOR_CHAINS.map((chainId: SupportedChainId) => (
          <ChainRow
            onSelectChain={async (targetChainId: SupportedChainId) => {
              await selectChain(targetChainId)
              toggleOpen()
            }}
            targetChain={chainId}
            key={chainId}
          />
        ))}
      </Column>
    </NavDropdown>
  )

  return (
    <Box position="relative" ref={ref}>
      <Row
        as="button"
        gap="8"
        className={styles.ChainSwitcher}
        background={isOpen ? 'accentActiveSoft' : 'none'}
        onClick={toggleOpen}
      >
        {!isSupported ? (
          <>
            <TokenWarningRedIcon fill={themeVars.colors.darkGray} width={24} height={24} />
            <Box as="span" className={subhead} display={{ sm: 'none', xl: 'flex' }} style={{ lineHeight: '20px' }}>
              Unsupported
            </Box>
          </>
        ) : (
          <>
            <img src={info.logoUrl} alt={info.label} className={styles.Image} />
            <Box as="span" className={subhead} display={{ sm: 'none', xl: 'flex' }} style={{ lineHeight: '20px' }}>
              {info.label}
            </Box>
          </>
        )}
        {isOpen ? (
          <NewChevronUpIcon width={16} height={16} color="blackBlue" />
        ) : (
          <NewChevronDownIcon width={16} height={16} color="blackBlue" />
        )}
      </Row>
      {isOpen && (isMobile ? <Portal>{dropdown}</Portal> : <>{dropdown}</>)}
    </Box>
  )
```

Reply:
There's no where in `isChainAllowed` that explicitly enforces this condition, which is why if this check is removed we have to do null checks on all the info fields. I've added what this would look like. Right now it is possible that a Supported Chain is added without info which would then cause a a half loaded state where no icon or label would be shown.